Add SwiftUI previews for widgets

samsymons opened the pull request:
CC: @brindy 

**Description**:

This PR adds previews of the widgets added in the iOS 14 update. Each widget has been added with both dark and light mode variants, and the widgets are populated with mock data to show that the randomized favourite color is working correctly. Here's a peek at a few of the widgets:

<img width="309" alt="widgets" src="https://user-images.githubusercontent.com/183774/93290896-9a0e3980-f796-11ea-8cb6-ccc79d68b47c.png">

There is a minor scheme change in this PR which was needed for SwiftUI previews to work. Before this change, the unit test target was being built when the main `DuckDuckGo` scheme builds, which caused compile errors when previews were displayed.

The solution to the compile error issue is to move the unit test config into its own scheme and wire it up to the `DuckDuckGo` scheme so that Command+U works as it always has. I also unchecked the `Show` option in the new `UnitTest` scheme so that it doesn't clutter up the scheme dropdown, since it can just be run from the main app scheme like always – from a dev work perspective, nothing should have changed.

Here's a before and after of the build log to prove that test targets are no longer being built during normal builds:

**Before:**

<img width="1441" alt="build-process-before" src="https://user-images.githubusercontent.com/183774/93290524-c7a6b300-f795-11ea-8b56-5f110b96e008.png">

**After:**

<img width="1441" alt="build-process-after" src="https://user-images.githubusercontent.com/183774/93290540-ce352a80-f795-11ea-9c84-2886c0f52819.png">

**Steps to test this PR**:

1. Open this branch in Xcode 12
1. Go to the `WidgetViews.swift` file and open the Canvas pane to verify that SwiftUI previews are displayed

**OS Testing**:

* [x] iOS 14

**Theme Testing**:

* [x] Light theme
* [x] Dark theme

samsymons committed:
Split the unit tests scheme out of the main app scheme.

This is so that the DuckDuckGo build action does not build unit tests during regular builds. This has been fine so far, but appears to trip up SwiftUI previews.

samsymons committed:
Add dark and light mode previews for each widget.

samsymons committed:
Add a bit of padding in the previews struct.

brindy reviewed (approved):
Wonder if we could have some showing the preview state as well, but I don't mind just now. 

LGTM! Thanks @samsymons

samsymons committed:
Add previews of the entries which have favorites.

samsymons committed:
Mark the search bar as unredacted.

The widget selection menu on iOS will sometimes take a second or two to load, and will mark the widget as redacted while it does so.

This is useful for sections of the widget with dynamic content, but the search bar does not need to be included.

brindy reviewed (commented):
LGTM!
